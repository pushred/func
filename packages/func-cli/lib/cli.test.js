const { resolve } = require('path');

const Cli = require('./cli');
const meow = require('meow');
const logger = require('./logger');

const CONFIG_PATH = resolve(__dirname, '../../../test');
const HELP = 'help!';

describe('Cli', () => {
  beforeEach(() => {
    process.argv = [];
  });

  test('returns a meow object', () => {
    return Cli(HELP, { searchFrom: CONFIG_PATH })
      .then(({ cli }) => {
        expect(Object.keys(cli))
          .toEqual(expect.arrayContaining(['input', 'flags', 'help']));
      });
  });

  test('loads config specified by flag', () => {
    process.argv = ['/usr/local/bin/node', 'cli.js', '--config', 'package.json'];

    return Cli(HELP, { searchFrom: CONFIG_PATH })
      .then(({ config }) => {
        expect(Object.keys(config))
          .toEqual(expect.arrayContaining(['func', 'colors', 'classes']));
        });
  });

  test('searches for config files', () => {
    return Cli(HELP, { searchFrom: CONFIG_PATH })
      .then(({ config }) => {
        expect(Object.keys(config))
          .toEqual(expect.arrayContaining(['func', 'colors', 'classes']));
        });
  });

  test('rejects if no config files are found', () => {
    return Cli(HELP)
      .catch(err => expect(err.message).toEqual('No configuration found'));
  });

  test('returns loaded config files', () => {
    return Cli(HELP, { searchFrom: CONFIG_PATH })
      .then(({ config }) => {
        const jsonConfig = JSON.parse(
          JSON.stringify(config)
            .split(CONFIG_PATH)
            .join('/var/tmp')
        );

        expect(jsonConfig).toMatchSnapshot()
      });
  });

  test('returns cosmiconfig search results', () => {
    return Cli(HELP, { searchFrom: CONFIG_PATH })
      .then(({ cli, configFiles }) => {
        expect(configFiles[0].includes('classes.yml'));
        expect(configFiles[1].includes('colors.yml'));
      });
  });
});
