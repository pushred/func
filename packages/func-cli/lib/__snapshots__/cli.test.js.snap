// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cli returns loaded config files 1`] = `
Object {
  "basePath": "/var/tmp",
  "classes": Object {
    "bg-button": "blue",
    "bg-gray": "gray",
    "bg-light-gray": "light-gray",
    "bg-lighter-gray": "lighter-gray",
    "button": "green",
    "error": "red",
  },
  "colors": Object {
    "blue": Object {
      "hue": 240,
      "lightness": "50%",
      "saturation": "100%",
    },
    "gray": Object {
      "hue": 180,
      "lightness": "0%",
      "saturation": "0%",
      "tints": 3,
    },
    "green": Object {
      "hue": 120,
      "lightness": "30%",
      "saturation": "50%",
    },
    "red": Object {
      "hue": 360,
      "lightness": "50%",
      "saturation": "100%",
    },
  },
  "func": Object {
    "files": Object {
      "classes": "classes.yml",
      "colors": "colors.yml",
    },
  },
  "paths": Object {
    "indexJson": "/var/tmp/func-index.json",
    "json": "/var/tmp/func.json",
    "stylesheet": "/var/tmp/func.css",
    "tokens": "/var/tmp/func-tokens.json",
  },
}
`;
